libraries/chain/controller.cpp

```diff
@@ -946,6 +946,8 @@ struct controller_impl {
    authorization_manager           authorization;
    protocol_feature_manager        protocol_features;
    controller::config              conf;
+   // write out chain_head after vote_processor has stopped to avoid concurrent access while writing
+   fc::scoped_exit<std::function<void()>> write_chain_head = [&]() { chain_head.write(conf.state_dir / config::chain_head_filename); };
```

Comment:
Maybe add a comment stating that this member has to come after `chain_head`, `fork_db` and `conf`.

Reply:
doesn't it depend on fork_db as well?

Reply:
In what way does it depend on forkdb?

Reply:
I thought [this](https://github.com/AntelopeIO/spring/blob/c2cffd8d7c255e997a0565fbebf3596298f16c5b/libraries/chain/include/eosio/chain/block_handle.hpp#L44) would write the `block_state` which is stored in fork_db?

Reply:
The block_state is stored in a shared_ptr that is referenced in forkdb. But the forkdb is not modifying the block state.

Reply:
I don't know what I was thinking. Thanks!